### PR TITLE
feat(dashboard): Widget Viewer persistent pagination

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Moves pagination values outside of state and onto router location so that it can be shared via url.
Also forces pagination to reset when changing query or sorting

### CODE REVIEW

File: static/app/components/modals/widgetViewerModal.tsx [hunk 1]
```diff
@@ -192,7 +190,12 @@ function WidgetViewerModal(props: Props) {
             onChange={(option: SelectValue<number>) =>
               router.replace({
                 pathname: location.pathname,
-                query: {...location.query, [WidgetViewerQueryField.QUERY]: option.value},
+                query: {
+                  ...location.query,
+                  [WidgetViewerQueryField.QUERY]: option.value,
+                  [WidgetViewerQueryField.PAGE]: undefined,
+                  [WidgetViewerQueryField.CURSOR]: undefined,
```
Comment (priscilawebdev) [inline static/app/components/modals/widgetViewerModal.tsx:197]: Usually, we use `cursor` in the URL to define which page we would like the UI to display and it is not clear to me why we need page too

**Comment by priscilawebdev** [comment]
I noticed that when we click on the sort, the cursor resets... is this something we expect?

**Comment by priscilawebdev** [comment]
https://user-images.githubusercontent.com/29228205/158135584-713ed18b-f1f1-4d92-8566-3594f0a18d0d.mov

**Comment by edwardgou-sentry** [comment]
> I noticed that when we click on the sort, the cursor resets... is this something we expect?

Hey Priscila, yup this is an expected outcome. Both Discover and Issues pages have this same behaviour when sorting so I've updated the Widget Viewer to match.

File: static/app/components/modals/widgetViewerModal.tsx (hunk 1 above)
Comment (edwardgou-sentry) [reply static/app/components/modals/widgetViewerModal.tsx:197]: The `cursor` for Issues API can sometimes indicate that a previous page exists even when there shouldn't be a previous page. We need to use a separate `page` value to work around this. Both `cursor` and `page` query parameters are used in the Issues page as well

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Pull request merged**